src/components/Animation/Controllers.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { lazy, useContext, useEffect, useState } from 'react'
+import { Spiral, SpiralFlyingLines } from '../../shared/2d/animations'
+import DataContext, { IDataContext } from '../Context'
+import { IAnimationProps } from './index'
+
+const RadiusCounter = lazy(() => import('../RadiusCounter'))
+const ParticlesCounter = lazy(() => import('../ParticlesCounter'))
+const VelocityCounter = lazy(() => import('../VelocityCounter'))
+const LineLengthCounter = lazy(() => import('../LineLengthCounter'))
+
+type Props = {
+  changeParticlesCount: any,
+  changeRadius: any,
+  changeVelocity: any,
+  changeLineLength: any
+} // which types do we need here
+
+const Controllers: React.FC<IAnimationProps & Props> = ({
+	properties,
+	classId,
+	changeParticlesCount,
+	changeRadius,
+	changeVelocity,
+	changeLineLength,
+}) => {
+	const [currentRange, setCurrentRange] = useState(0)
+	const { keyToggle } = useContext<IDataContext>(DataContext)
+
+	useEffect(() => {
+		const items = document.querySelector('.animation-handlers')
+
+		if (items) {
+			items.children[currentRange].classList.add('active')
+		}
+
+		const handleOnClick = (event: React.MouseEvent<HTMLElement>) => {
+			const target = event.target as Element
+
+			if (event.currentTarget) {
+				const childs = [...event.currentTarget.children]
+				childs.forEach((child) => {
+					child.classList.remove('active')
+				})
+
+				if (target.className === 'count' || target.className === 'counter') {
+					target.parentElement?.classList.add('active')
+					setCurrentRange(childs.findIndex(item => item.className.includes('active')))
+					return
+				}
+
+				target.classList.add('active')
+				setCurrentRange(childs.findIndex(item => item.className.includes('active')))
+			}
+		}
+
+		const handleOnMouseOver = (event: React.MouseEvent<HTMLElement>) => {
+			const target = event.target as Element
+
+			if (event.currentTarget) {
+				const childs = [...event.currentTarget.children]
+				childs.forEach((child) => child.classList.remove('active'))
+
+				if (target.className === 'count' || target.className === 'counter') {
+					target.parentElement?.classList.add('active')
+					return
+				}
+
+				target.classList.add('active')
+			}
+		}
+
+		const handleOnMouseOut = (event: React.MouseEvent<HTMLElement>) => {
+			if (event.currentTarget && items) {
+				const childs = [...event.currentTarget.children]
+				childs.forEach((child) => child.classList.remove('active'))
+				items.children[currentRange].classList.add('active')
+			}
+		}
+
+		if (items) {
+			items.addEventListener('click', handleOnClick)
+
+			items.addEventListener('mouseover', handleOnMouseOver)
+
+			items.addEventListener('mouseout', handleOnMouseOut)
+		}
+
+		return () => {
+			items.removeEventListener('click', handleOnClick)
+
+			items.removeEventListener('mouseover', handleOnMouseOver)
+
+			items.removeEventListener('mouseout', handleOnMouseOut)
+		}
+	}, [currentRange, classId])
```

Comment:
So. I won't try to comment on each line but give you some advice.

1. If you create an element with a specific logic (open and close by hover and click) it will be good to put this element in a separate component. By this solution, you wouldn't need to use inefficient React DOM requests.
2. If you have multiple same components with different props, use the `.map` function and the `key` prop.  
3. Store callback outside the `useEffect` in `useCallbacks`.
4. Of course, you need a main state for defining what component is now active. You can have it in the current component. But **important** that classnames need to depend on state. By event handler you need to change state, and by state changed you need to change className.